ui/src/app/views/workspace-edit.tsx

```diff
@@ -360,6 +360,148 @@ export const LabeledCheckBox = (props) => {
+  START: number;
+  ACTIVE: number;
+  SUGGEST: number;
+  HOVER: number;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      matches: [],
+      hover: [],
+      state: this.START,
+    };
+    this.searchTermChangedEvent = fp.debounce(300, this._search);
+
+    // State machine states
+    this.START = 0;   // Input box lacks focus, drop-down invisible
+    this.ACTIVE = 1;  // Input box focused, drop-down invisible
+    this.SUGGEST = 2; // Input box focused, search results visible
+    this.HOVER = 3;   // Input box focused, search results visible, item activated
+  }
+
+  _toggleHover(j) {
+    return this.state.hover.map((elt, i) => {
+      return (i === j) ? !elt : elt;
+    });
+  }
+
+  _search(keyword: string): void {
+    this.setState({matches: []});
+    keyword = keyword.trim();
+    if (!keyword) {
+      return;
+    }
+    this.props.onSearch(keyword).then((matches) => {
+      this.setState({
+        matches: matches,
+        state: matches.length > 0 ?
+          this.SUGGEST : this.ACTIVE,
+        hover: matches.map(() => false),
+      });
+    }).catch((e) => {});
+  }
+
+  _onBlur() {
+    const state = this.state.state;
+    if (state === this.ACTIVE  ||
+        state === this.SUGGEST ||
+        state === this.HOVER) {
+      this.setState({state: this.START});
+    }
+  }
+
+  _onChange(match) {
+    this.setState({state: this.ACTIVE});
+    this.searchTermChangedEvent(match);
+    this.props.onChange(match);
+  }
+
+  _onFocus() {
+    this.setState({state: this.ACTIVE});
+  }
+
+  _onMouseOver(j) {
+    this.setState({
+      state: this.HOVER,
+      hover: this._toggleHover(j)
+    });
+  }
+
+  _onMouseOut(j) {
+    this.setState({
+      state: this.SUGGEST,
+      hover: this._toggleHover(j)
+    });
+  }
+
+  _onMouseDown(match) {
+    this.props.onChange(match);
+  }
+
+  _suggest() {
+    const s = this.state.state;
+    const rval = this.props.enabled &&
+      (s === this.SUGGEST || s === this.HOVER);
+    return rval;
+  }
+
+  render() {
+    return (
+      <div style={{position: 'relative'}}>
+        <TooltipTrigger
+          content={this.props.tooltip}
+          disabled={this.props.enabled}>
+          <TextInput
+            value={this.props.value}
+            style={{width: '90%', border: '1px solid #9a9a9', borderRadius: '5px'}}
+            onFocus={this._onFocus.bind(this)}
+            onBlur={this._onBlur.bind(this)}
+            onChange={e => this._onChange(e)}
+            placeholder={this.props.placeholder}
+            disabled={!this.props.enabled}/>
+        </TooltipTrigger>
+        {this._suggest() &&
+          <div data-test-id='drop-down'
+            style={{...styles.dropdownMenu, ...styles.open, minWidth: '90%'}}>
+            {this.state.matches.map((disease, j) => {
+              return (
+                <div key={j} style={this.state.hover[j] ? styles.boxHover : styles.box}
+                  onMouseOver={this._onMouseOver.bind(this, j)}
```

Comment:
If its not going from angular to react, you may be able to do this as `(j) => this._onMouseOver(j)` (And similar to below)

Reply:
`j` is bound, though i suppose i could do `() => this._onMouseOver(j)`. tomayto/tomahto